pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -178,22 +177,23 @@ func (l *listUsersQuery) ListUsers(
 	}()
 
 	go func() {
-		defer close(foundUsersCh)
 		internalRequest := fromListUsersRequest(req, &datastoreQueryCount)
-		if err := l.expand(cancellableCtx, internalRequest, foundUsersCh); err != nil {
-			expandErrCh <- err
-			return
-		}
+		err := l.expand(cancellableCtx, internalRequest, foundUsersCh)
+		close(foundUsersCh)
+		expandErrCh <- err
 	}()
 
-	select {
-	case err := <-expandErrCh:
-		telemetry.TraceError(span, err)
-		return nil, err
-	case <-done:
-		cancelContextIfMaxResultsMet()
-		break
+	err := <-expandErrCh
+	if err != nil {
+		if !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
One thing I want/need to add a test for here as well. If we're getting `context.DeadlineExceeded` from the storage layer, then I want to treat it differently.

I will add a test for this to make sure we're treating it properly. I think the way the code is written right now it will skip the internal error.

Reply:
Reminder – if we revert to a swtich (as mentioned above), we still need to nil-check the err before tracing it.

Reply:
Adjust the control flow. Now we only send on the channel if `if err != nil`